src/Bicep.Core/Parsing/Parser.cs

```diff
@@ -286,19 +282,26 @@ TokenType.StringLeftPiece or
 
         private ProviderDeclarationSyntax ProviderImportDeclaration(Token keyword, IEnumerable<SyntaxBase> leadingNodes)
         {
-            var providerSpecification = this.WithRecovery(
-                () => ThrowIfSkipped(this.InterpolableString, b => b.ExpectedProviderSpecification()),
-                RecoveryFlags.None,
-                TokenType.Assignment,
-                TokenType.NewLine);
+            var providerSpecificationSyntax = reader.Peek().Type switch
+            {
+                TokenType.Identifier => this.WithRecovery(
+                    () => IdentifierOrSkip(b => b.ExpectedProviderAliasName()),
+                    RecoveryFlags.None,
+                    TokenType.NewLine),
+
+                _ => this.WithRecovery(
+                    () => ThrowIfSkipped(this.InterpolableString, b => b.ExpectedProviderSpecification(false)),
```

Comment:
It's innaccurate to pass `false` here. My recommendation would be to simplify the diagnostic to just show all possible syntax forms, and not bother trying to do feature detection at parse time.